Source/Core/Core/Debugger/PPCDebugInterface.cpp

```diff
@@ -383,10 +384,36 @@ void PPCDebugInterface::RunToBreakpoint()
 {
 }
 
+std::shared_ptr<Core::NetworkCaptureLogger> PPCDebugInterface::NetworkLogger()
+{
+  const bool has_ssl = Config::Get(Config::MAIN_NETWORK_SSL_DUMP_READ) ||
+                       Config::Get(Config::MAIN_NETWORK_SSL_DUMP_WRITE);
+  const auto current_capture_type =
```

Comment:
It's possible I'm missing something but I feel like this might be a little bit simpler:

```
if (has_ssl)
{
  if (!m_network_logger || m_network_logger->GetCaptureType() != Core::NetworkCaptureType::Raw)
    m_network_logger = std::make_shared<Core::BinarySSLCaptureLogger>();
 }
 else
{
  if (!m_network_logger || m_network_logger->GetCaptureType() != Core::NetworkCaptureType::None)
    m_network_logger = std::make_shared<Core::DummyNetworkCaptureLogger>();
}

return m_network_logger;
```

Reply:
This function acts a bit like a singleton pattern, if the logger isn't initialized or has been changed it creates a new one.

The alternative you provided doesn't sound very scalable to me if you have more logging types (ex: the PCAP one will be in a follow up PR). Your alternative also call `GetCaptureType()` multiple times.

Reply:
Ah ok, sorry, I didn't realize there were more types planned.  I'm curious though, how will that work with the current config.  There's only a single boolean for ssl at the moment.

I see what you're trying to go for but I personally feel like the `reset()` then call `make_shared` in the same pass, along with multiple checks to do so is quite odd.  That's why I was trying to suggest an alternative.

Maybe a simple change would suffice:

```
if (m_network_logger && m_network_logger->GetCaptureType() == current_capture_type)
  return m_network_logger;

switch (current_capture_type)
{
  case Core::NetworkCaptureType::Raw:
    m_network_logger = std::make_shared<Core::BinarySSLCaptureLogger>();
    break;
  case Core::NetworkCaptureType::None:
    m_network_logger = std::make_shared<Core::DummyNetworkCaptureLogger>();
    break;
};

return m_network_logger;
```

Thoughts?

Reply:
Yes, you have a point. Inverting the condition is way easier to read. I'm going to modify that, ASAP.